examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,289 @@
+
+First of all, we would like to thank Professor Johnathan Pillow
+and Professor E.J. Chichilnisky for sharing their tutorial dataset.
+The original MATLAB and Python tutorial can be found from
+https://github.com/pillowlab/GLMspiketraintutorial.
+
+These data were collected by Valerie Uzzell in the lab of
+E.J. Chichilnisky at the Salk Institute. For full information see
+Uzzell et al. (J Neurophys 04) [1]_, or Pillow et al. (J Neurosci 2005) [2]_.
+
+In this tutorial, we will demonstrate how to fit linear GLM and
+linear-nonlinear GLM (a.k.a Poisson GLM) to predict the spike counts
+recorded from primate retinal ganglion cells.
+The dataset contains spike responses from 2 ON and 2 OFF parasol
+retinal ganglion cells (RGCs) in primate retina, stimulated with
+full-field `binary white noise`. Two experiment performed consisted of a
+long (20-minute) binary stochastic (non-repeating) stimulus
+which can be used for computing the spike-triggered average
+(or characterizing some other model of the response).
+
+References
+----------
+.. [1] Uzzell, V. J., and E. J. Chichilnisky. "Precision of
+   spike trains in primate retinal ganglion cells."
+   Journal of Neurophysiology 92.2 (2004)
+.. [2] Pillow, Jonathan W., et al. "Prediction and decoding of
+   retinal ganglion cell responses with a probabilistic
+   spiking model." Journal of Neuroscience 25.47 (2005)
+
+"""
+
+# Authors: Titipat Achakulvisut <my.titipat@gmail.com>
+# License: MIT
+
+########################################################
+#
+# Import all the relevance libraries.
+
+import os.path as op
+import json
+
+import numpy as np
+from numpy.linalg import inv
+from scipy.linalg import hankel
+
+from pyglmnet import GLM
+
+import matplotlib.pyplot as plt
+from tempfile import TemporaryDirectory
+
+########################################################
+#
+# Fetch the dataset. The JSON file contains the
+# following keys:  ``stim`` (binary stochastic stimulation),
+# ``stim_times`` (time of the stimulation), and
+# ``spike_times`` (recorded time of the spikes)
+
+from pyglmnet.datasets import fetch_RGCs_data
+
+with TemporaryDirectory(prefix="tmp_glm-tools") as temp_dir:
+    dpath = fetch_RGCs_data(dpath=temp_dir, accept_rgcs_license=True)
+    with open(op.join(dpath, 'data_RGCs.json'), 'r') as f:
+        rgcs_dataset = json.loads(f.read())
+
+stim = np.array(rgcs_dataset['stim'])
+stim_times = np.array(rgcs_dataset['stim_times'])
+
+# spike times for all 4 cells (0-1 are OFF cells, 2-3 are ON cells)
+spike_times = [
+    np.array(rgcs_dataset['spike_times']['cell_0']),
+    np.array(rgcs_dataset['spike_times']['cell_1']),
+    np.array(rgcs_dataset['spike_times']['cell_2']),
+    np.array(rgcs_dataset['spike_times']['cell_3']),
+]
+
+n_cells = len(spike_times)  # total number of cells
+dt = stim_times[1] - stim_times[0]  # time between the stimulation
+n_t = len(stim)  # total number of the stimulation
+f = 1. / dt  # frequency of the stimulation
```

Comment:
single letter variable names are not good, you will have trouble doing a Ctrl + F on them. Plus they are not very descriptive.

```suggestion
sfreq = 1. / dt  # frequency of the stimulation
```